plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -1270,11 +1270,24 @@ public static function get_subscriptions() {
 			return $data;
 		}
 
+		$source = $_SERVER['REQUEST_URI'];
+		if ( preg_match( '/\S+wc-addons?\S+/', $source ) ):
+			$source = 'my-subscriptions';
+		elseif ( preg_match( '/\S+plugins.php?\S+/', $source ) ):
+			$source = 'plugins';
+		elseif ( preg_match( '/\S+wc-admin?\S+/', $source ) ):
+			$source = 'inbox-notes';
+		elseif ( preg_match( '/\Sadmin-ajax.php/', $source ) ):
+			$source = 'heartbeat-api';
+		endif;
```

Comment:
I wonder if `stripos` would work for some or all of these conditions? `preg_match` is exact, but a bit slow, and we're doing it several times here.